lib/intelligent_foods/api_client.rb

```diff
@@ -34,8 +34,13 @@ def execute_request(request:, uri:, body: nil,
       end
     end
 
-    def build_request_with_body(uri:, body:)
-      request = Net::HTTP::Post.new(uri)
+    def build_patch_request(uri:, body:)
+      build_request_with_body(uri: uri, body: body,
+                              http_method: Net::HTTP::Patch)
+    end
```

Comment:
This method isnt doing much, is there a reason you want to avoid using `build_request_with_body` directly and pass `http_method` in the order update?

Reply:
@imRohan your instinct was correct. take a look at what i have sketched out [here](https://github.com/Intelligent-Foods/intelligent-foods-ruby/compare/main...cw/auth). bring back your patch method and mimic this as closely as is reasonable within the scope of this PR